src/Model/Donor.php

```diff
@@ -50,10 +52,88 @@ class Donor
      */
     private $mandateSource;
 
-    public function __construct(DonorState $state, string $mandateSource)
-    {
+    /**
+     * @var string
+     */
+    private $payerNumber;
+
+    /**
+     * @var \byrokrat\banking\AccountNumber
+     */
+    private $account;
+
+    /**
+     * @var \byrokrat\id\Id
+     */
+    private $id;
+
+    /**
+     * @var string
+     */
+    private $comment;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var \byrokrat\giroapp\Model\PostalAddress
+     */
+    private $address;
+
+    /**
+     * TODO:
+     * Contact information should be moved to a separate object, ContactPerson.
+     * It should be clear in the end application that this contact information,
+     * if added, should only be used to contact the donor on autogiro subjects.
+     * This to conform with Data protection regulations.
+     */
+
+    /**
+     * @var string
+     */
+    private $email;
+
+    /**
+     * @var string
+     */
+    private $phone;
+
+    /**
+     * @var SEK
+     */
+    private $donationAmount;
+
+    /**
+     * $var string
+     */
+    private $uid;
+
+    public function __construct(
+        DonorState $state,
+        string $mandateSource,
+        string $payerNumber,
+        \byrokrat\banking\AccountNumber $account,
+        \byrokrat\id\Id $id,
+        string $name,
+        PostalAddress $address = null,
+        SEK $donationAmount = null,
+        string $comment = ""
+    ) {
         $this->setState($state);
         $this->mandateSource = $mandateSource;
+        $this->payerNumber = $payerNumber;
+        $this->account = $account;
+        $this->id = $id;
+        $this->comment = $comment;
+        $this->name = $name;
+
+        $this->email = "";
+        $this->phone = "";
+        $this->address = $address ?: new PostalAddress();
+        $this->donationAmount =  $donationAmount ?: new SEK('0');
+        $this->uid = hash('sha256', $this->id->format('S-sk').$this->account->getNumber());
```

Comment:
Should actually use account->get16(). This is because swedbank account numbers may contain a clearing number check digit. Since this check digit is optional we want the same account number to yield the same hash regardless if a clearing checkdigit is used. get16() ensures this..

$id->format('S-sk') is perfect. Good catch!